code/mission/missionlog.cpp

```diff
@@ -46,139 +34,27 @@
-	// If so, we can remove any subsystem destroyed entries from the log for this ship.  
-	if ( type == LOG_SHIP_DESTROYED || type == LOG_SELF_DESTRUCTED ) {
-		for (i = 0; i < last_entry; i++) {
-			entry = &log_entries[i];
-
-			// check to see if the type is a subsystem destroyed entry, and that it belongs to the
-			// ship passed into this routine.  If it matches, mark as obsolete.  We'll clean up
-			// the log when it starts to get full
-			if ( !stricmp( pname, entry->pname ) ) {
-				if ( (entry->type == LOG_SHIP_SUBSYS_DESTROYED) || (entry->type == LOG_SHIP_DISARMED) || (entry->type == LOG_SHIP_DISABLED) )
-					entry->flags |= MLF_OBSOLETE;
-			}
-		}
-	}
-
-	// check to see if we are getting to about 80% of our log capacity.  If so, cull the log.
-	if ( last_entry > LOG_CULL_MARK ) {
-		mission_log_cull_obsolete_entries();
-
-		// if we culled the entries, and we are still low on space, we need to take more drastic measures.
-		// these include removing all non-essential entries from the log.  These entries are entries 
-		// which has not been asked for by mission_log_get_time
-		if ( last_entry > LOG_CULL_MARK ) {
-			nprintf(("missionlog", "marking the first %d non-essential log entries as obsolete\n", LOG_LAST_DITCH_CULL_NUM));
-			for (i = 0; i < LOG_LAST_DITCH_CULL_NUM; i++ ) {
-				entry = &log_entries[i];
-				if ( !(entry->flags & MLF_ESSENTIAL) ){
-					entry->flags |= MLF_OBSOLETE;
-				}
-			}
-
-			// cull the obsolete entries again
-			mission_log_cull_obsolete_entries();
-
-			// if we get to this point, and there are no entries left -- we are in big trouble.  We will simply
-			// mark the first 20% of the log as obsolete and compress.  Don't do this unless we are *really*
-			// in trouble
-			if ( last_entry > LOG_CULL_DOORDIE_MARK ) {
-				nprintf(("missionlog", "removing the first %d entries in the mission log!!!!\n", LOG_LAST_DITCH_CULL_NUM));
-				for (i = 0; i < LOG_LAST_DITCH_CULL_NUM; i++ ){
-					entry->flags |= MLF_OBSOLETE;
-				}
-
-				mission_log_cull_obsolete_entries();
-			}
-		}
-	}
+	Log_entries.clear();
 }
 
 // following function adds an entry into the mission log.
 // pass a type and a string which indicates the object
 // that this event is for.  Don't add entries with this function for multiplayer
 void mission_log_add_entry(LogType type, const char *pname, const char *sname, int info_index, int flags)
 {
-	__UNUSED int last_entry_save;
-	log_entry *entry;	
-
 	// multiplayer clients don't use this function to add log entries -- they will get
 	// all their info from the host
 	if ( MULTIPLAYER_CLIENT ){
 		return;
 	}
 
-	last_entry_save = last_entry;
-
-	// mark any entries as obsolete.  Part of the pruning is done based on the type (and name) passed
-	// for a new entry
-	mission_log_obsolete_entries(type, pname);
-
-	entry = &log_entries[last_entry];
-
-	if ( last_entry == MAX_LOG_ENTRIES ){
-		return;
-	}
+	Log_entries.emplace_back();
+	auto entry = &Log_entries.back();
```

Comment:
I'd love for this (and other auto-assigned types) to at least be an auto* instead for readability

Reply:
Interesting... I'm of the opposite opinion, that using `auto*` is less readable than simply using `auto`.  Unlike `const auto` or `auto &`, `auto *` doesn't change the meaning of the declaration here.